src/Gui/ActiveObjectList.cpp

```diff
@@ -197,3 +197,15 @@ void ActiveObjectList::objectDeleted(const ViewProviderDocumentObject &vp)
         }
     }
 }
+
+std::optional<App::DocumentObject*> ActiveObjectList::findActiveObjectWithExtension(const Base::Type& extensionTypeId) const
```

Comment:
* The pointer already express optional-ness (nullptr), this is also how `getObject` works
* How about naming it `getObjectWithExtension` instead?
* `Base::Type` is so small that it's recommended to copy it by value rather than by reference.
```suggestion
App::DocumentObject* ActiveObjectList::getObjectWithExtension(const Base::Type extensionTypeId) const
```

Reply:
Done, although I renamed it to `getActiveObjectWithExtension` to be explicit that it is an active object.